firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -4556,22 +4557,34 @@ public void testGetReturnsNullForEmptyNodeWhenOnline()
   }
 
   @Test
-  public void testGetThrowsExceptionForEmptyNodeWhenOffline()
-      throws DatabaseException, InterruptedException {
+  public void testGetWaitsForConnection() throws DatabaseException, InterruptedException {
     FirebaseApp app =
         appForDatabaseUrl(IntegrationTestValues.getNamespace(), UUID.randomUUID().toString());
     FirebaseDatabase db = FirebaseDatabase.getInstance(app);
     DatabaseReference node = db.getReference().child(db.getReference().push().getKey());
+    long offlinePause = 3000L;
     db.goOffline();
+    THREAD_POOL_EXECUTOR.execute(
+        () -> {
+          try {
+            Thread.sleep(offlinePause);
+          } catch (InterruptedException e) {
+            fail("Exception while pausing for get.");
+          } finally {
+            db.goOnline();
+          }
+        });
     try {
+      long startTime = System.currentTimeMillis();
       Tasks.await(node.get());
+      long elapsed = System.currentTimeMillis() - startTime;
+      // Since get starts after we schedule the re-connect task, it could technically
+      // block for slightly less long than offlinePause, but the difference should be
+      // negligible.
+      assertTrue(elapsed > offlinePause * 0.75);
```

Comment:
This seems like it has the potential to cause flakes and does not actually add a lot of value. How about doing something like:


```
AtomicBoolean receivedValue = new AtomicBoolean();
  THREAD_POOL_EXECUTOR.execute(
        () -> {
          try {
            Thread.sleep(offlinePause);
          } catch (InterruptedException e) {
            fail("Exception while pausing for get.");
          } finally {
            assertFalse(receivedValue.get());
            db.goOnline();
          }
        });
    try {
      Tasks.await(node.get());
      receivedValue.set(true);
```